esmvaltool/_task.py

```diff
@@ -134,18 +134,28 @@ def _format(value):
         return txt
 
     def _format_dict(name, dictionary):
-        """Format dict as NCL"""
-        lines = ['{} = True'.format(name)]
+        """Format dict as NCL (list of) logical(s) with attributes."""
+        if name in ('diag_script_info', 'config_user_info'):
```

Comment:
The purpose of this function is convert a generic python dict/yaml mapping to NCL. Variable specific re-ordering etc so results are valid ncl, should be done in the function `esmvaltool.preprocessor._io._write_ncl_metadata`, as described here: https://github.com/ESMValGroup/ESMValTool/issues/531#issuecomment-433075375. I think it would be best not to change this.

Reply:
Ok. This was just to distinguish the case where a list is needed from the ones where a scalar is enough. But I can do this in `interface.ncl` and leave the python function more generic.

Reply:
it's fine if you do it in Python too, but do it in `esmvaltool.preprocessor._io._write_ncl_metadata` and not here, that keeps a good separation between translating a structure of lists/dicts/variables/etc to ncl and constructing a particular structure for a particular purpose.

I think just removing these lines should already help:
https://github.com/ESMValGroup/ESMValTool/blob/cfcb139a2e8263c9f0cc440b36c2fd3f0391dc50/esmvaltool/preprocessor/_io.py#L233-L236
because there we translate a list of dicts to a dict of lists, and that is no longer needed if we use the ncl list type.

Reply:
Now this is sorted out in `interface.ncl`, but let me have a try in `_io.py` as you suggest.

Reply:
I'm not sure how you want to change this. We are switching to NCL lists to support multiple variables here, but each element of the list is still a logical with attributes that translates the python dictionary.
Are you proposing to redefine this as a list of lists?

Reply:
No, not necessarily lists of lists. A list of logicals where each logical describes one preprocessed file. I've done the changes I propose in PR #692, please have a look.